### ISSUE TITLE
Evernote: the creation of developer tokens is temporarily disabled.

### ISSUE THREAD

**Issue by vincentfs** [bordaigorl/sublime-evernote#194]
So is there a workaround to continue using this great plugin?

See: 
https://www.evernote.com/api/DeveloperToken.action

**Comment by rdhyee** [comment]
Wow, that's unfortunate. I got the same error message as you this morning. I just submitted a ticket to Evernote Support to try to get an explanation of what's happening with dev tokens and when the situation will be fixed.

**Comment by z2x** [comment]
same as I, and how should we use it later?

**Comment by bordaigorl** [comment]
Oh no this is horrible. Let's hope they resume the service soon.
In the meantime I can try and submit a request for an App Key, which would mean nobody would need a dev token anymore. I am not sure the plugin fulfills their (undisclosed?) criteria for activating production keys.
I'll keep you posted.

**Comment by dbrits17** [comment]
If Evernote doesn't approve your request for an App Key, I will be devastated. I use Sublime and Evernote daily at work, and without this plugin to allow communication between the two, I'm going to have to rethink my entire workflow. This just sucks.

**Comment by alexcthompson** [comment]
Wow - this would mean no more Evernote for me if the issue persists = your repo is essential to my workflow as well (thank you). I've sent a ticket to Evernote to this effect.

**Comment by rdhyee** [comment]
&lt;rant>

I wrote filed a ticket almost a week ago but haven't gotten any response to the ticket.  I've also tweeted the CEO of Evernote about this matter: https://twitter.com/rdhyee/status/911342623461224449.  I'm considering writing him an email too.  Chris O'Neill has invited people to write to him personally about important Evernote-related matters.  From [A Note from Chris O'Neill](https://blog.evernote.com/blog/2015/09/29/a-note-from-chris-oneill/):

> This is an important moment in the life of our company. I want to thank all the Evernoters, past and present, for creating a product that is such a positive force in the lives of individuals and teams around the world. This company and this product are truly special, and I intend to keep it that way. I look forward to sharing more specific details with you in the months ahead. In the meantime, please send your thoughts to me directly at ceochris@evernote.com.

How dev tokens were shut down without any warning, explanation, or concrete public plans about next steps is deeply worrisome for anyone depending on Evernote to be reliable and trustworthy for the long haul.  See [Evernote CEO Chris O'Neill ridicules Google rival as a 'hobby' and suggests no-one loves Microsoft - Business Insider](http://www.businessinsider.com/evernote-ceo-chris-oneill-ridicules-google-rival-as-a-hobby-and-suggests-no-one-loves-microsoft-2017-3)

&lt;/rant>

**Comment by axfelix** [comment]
I just hit this yesterday... pretty bummed! Haven't gotten a response from them yet either.

**Comment by buginux** [comment]
I hit this just now, I just can't format my evernote article property without this plugin. This is unfortunate.

**Comment by Strangehill** [comment]
😱

**Comment by richardxu100** [comment]
I'm pretty frustrated by this too. I might need to switch to google docs if I can't use evernote with my text editor.

**Comment by rdhyee** [comment]
Evernote Helps [responded to me this morning](https://twitter.com/evernotehelps/status/911995726640664578):

> In the meantime you can go through the OAuth flow to obtain a token!  Apologies for the wait. /JG

I then [asked](https://twitter.com/rdhyee/status/912027969874386944):

> Which OAuth flow?  Should folks missing dev tokens get production API keys at https://dev.evernote.com/#  to do our own OAuth flow?

**Comment by bordaigorl** [comment]
So, I requested an API key, I got the sandboxed one. Before I request them to activate it on the production servers I wanted to test the OAuth flow on the sandboxed first. However I am very busy at the moment and will not be able to experiment on this until the weekend.

*PLEA*: if anybody feels like giving it a try let me know!

Another issue that bothers me a little: how does one go about storing the keys securely in a plain-text plugin like this?
I simply never wrote an OAuth app before so I need guidance on the security aspects on the plugin's side. Any suggestions/volunteers?

**Comment by alexcthompson** [comment]
So I got this email from Evernote Support just now:

```Jason C. (Evernote Help and Learning)
Sep 25, 12:59 PDT

Hi Alex,

We expect to have the developer token option re-enabled for you by the end of this week.

When it has been re-enabled you will be able to create a new one.

Thank you for your patience.

Best,
Jason```

**Comment by dbrits17** [comment]
I had to send a message to @EvernoteHelps on Twitter and this was their response:

"Thank you for reaching out. We don’t have an ETA at the moment, but in the meantime you can go through the OAuth flow to obtain a token! Apologies for the wait. 

Best, 

-Jillian"

**Comment by axfelix** [comment]
The page to request regular dev tokens is back for me!

**Comment by dbrits17** [comment]
The page is back for me too! Thanks for the heads up, axfelix. I was finally able to request a developer token.

**Comment by parenillo** [comment]
Still disabled for me :(
Hopefully it comes back soon since it's back for other people?

**Comment by ssk2** [comment]
Resolved for me too 😌 .

**Comment by richardxu100** [comment]
Unfortunately, it's still disabled for me too

**Comment by rdhyee** [comment]
The dev token has been re-enabled for me. Has everyone who doesn't have access to a dev token filed a support ticket?  It may or may not help since when I just looked at how to file a ticket, I see the error message:

> We're experiencing higher ticket volumes than normal. It may take up to one week before you receive a reply. For immediate help, fill out the fields below then click the "Chat now" button.

**Comment by dbrits17** [comment]
I don't have a paid Evernote account, so I wasn't even able to submit a ticket. I had to contact them on Twitter via @EvernoteHelps.

**Comment by pmarkaide** [comment]
Enabled for me and everything works as before :)

For the people that don't have access yet, I send a ticket on day one. Without answer, I've access to the dev token today. I don't think the activation is related to the ticket, but I think is recommended to send one as @rdhyee says.

**Comment by parenillo** [comment]
Enabled after submitting a ticket 3 days ago~

**Comment by z2x** [comment]
Unfortunately, it's still disabled for me

**Comment by vincentfs** [comment]
It seems that you can now submit ticket to get this resolved.

**Comment by arikalmen** [comment]
still not available in jan. 2018

**Comment by WarrenMarshall** [comment]
Still broken in Feb 2018 ... they aren't going to fix it are they?

**Comment by bordaigorl** [comment]
I think they enable it on a per-user basis.
Safest bet is to request a new token from https://www.evernote.com/SupportLogin.action
Their communication on the issue is awful.

**Comment by kingname** [comment]
Still not work in Feb 25 in 2018

**Comment by hsiaoyu-chou** [comment]
I sent a request via [https://www.evernote.com/SupportLogin.action](https://www.evernote.com/SupportLogin.action) two days ago, and now it is already enabled.

**Comment by Pierpa** [comment]
Still not work in Mar 01 in 2018

**Comment by mausv** [comment]
You just have to contact Evernote to enable the option, I got it enable within minutes after getting in chat with a representative.

**Comment by xdaudaudau** [comment]
Not every body can contact them...

**Comment by edoulesG** [comment]
Still not work in August 15th in 2018.
I lose my hope for this function already when i finish these commit

**Comment by kingname** [comment]
You can not use this function forever. give up.

**Comment by sneak** [comment]
I just reached out to them via the support chat and they quickly and without fuss re-enabled the ability to get developer tokens on both of my accounts.

**Comment by jacobbaratta** [comment]
I can confirm that Evernote support is still re-enabling this on a per-account basis upon request in 2021. I reached out to Evernote support via the Chat option after I found this issue, and I was able to get it enabled for my account in literally 3 minutes flat. All I had to provide was my Evernote account email address (they still don't require justification for the developer API, unlike the OAuth API).
